plugin/config_test.go

```diff
@@ -238,6 +239,206 @@ pipeline:
 	}
 }
 
+type PluginOutputIDTestCase struct {
+	Name          string
+	PluginConfig  pipeline.Config
+	ExpectedOpIDs map[string][]string
+}
+
+func TestPluginOutputIDs(t *testing.T) {
+	// TODO: ids shouldn't need to be specified once autogen IDs are implemented
+	pluginContent := []byte(`
+parameters:
+pipeline:
+  - id: noop
+    type: noop
+  - id: noop1
+    type: noop
+    output: {{ .output }}
+`)
+	pluginName := "my_plugin"
+	pluginVar, err := NewPlugin(pluginName, pluginContent)
+	require.NoError(t, err)
+	operator.RegisterPlugin(pluginVar.ID, pluginVar.NewBuilder)
+
+	// TODO: remove ID assignment
+	pluginContent2 := []byte(`
+parameters:
+pipeline:
+  - id: noop
+    type: noop
+  - id: noop1
+    type: noop
+`)
+	secondPlugin := "secondPlugin"
+	secondPluginVar, err := NewPlugin(secondPlugin, pluginContent2)
+	require.NoError(t, err)
+	operator.RegisterPlugin(secondPluginVar.ID, secondPluginVar.NewBuilder)
+
+	cases := []PluginOutputIDTestCase{
+		{
+			Name: "same_op_outside_plugin",
+			PluginConfig: func() []operator.Config {
+				return pipeline.Config{
+					operator.Config{
+						// TODO: ID should be noop to start then auto gened to noop2
```

Comment:
This is not accurate. Due to namespacing, IDs that exist within plugins should not collide with IDs at other levels or within other plugins.

This first test case should work without specifying the ID of the first operator:
```yaml
pipeline:
  - type: noop
  - type: my_plugin

# my_plugin
pipeline:
  - type: noop
    id: noop
  - type: noop
    id: noop1
    output: {{ .output }}
```

The expected pipeline would be:
`$.noop` => `$.my_plugin.noop` => `$.my_plugin.noop1`

---

Since we're also testing that the plugin ID is set correctly, let's either enhance this test or add another one that effectively does:

```yaml
pipeline:
  - type: noop
  - type: my_plugin
  - type: noop1

# my_plugin
pipeline:
  - type: noop
    id: noop
  - type: noop
    id: noop1
    output: {{ .output }}
```

The expected pipeline would be:
`$.noop` => `$.my_plugin.noop` => `$.my_plugin.noop1` => `$.noop1`

Reply:
Enhanced the test to better show scoping works.